### PR TITLE
Flow typings

### PR DESCRIPTION
**Pull request by Ragnar-H**
## Status
**Ready**

## Description
Static type checking of what we got.
CI now lints and flow checks

## Example | Screenshots
Getting rid of type errors such as:
```
src/Components/CaseCard/index.js:30
 30:         subtitle={parliament_case.parliament_session.name}
                                       ^^^^^^^^^^^^^^^^^^ property `parliament_session`. Property not found in
 30:         subtitle={parliament_case.parliament_session.name}
                       ^^^^^^^^^^^^^^^ object type

src/Components/CaseCard/index.js:38
 38:             <span>Parliament: {parliament_case.parliament_session.parliament.name}</span>
                                                    ^^^^^^^^^^^^^^^^^^ property `parliament_session`. Property not found in
 38:             <span>Parliament: {parliament_case.parliament_session.parliament.name}</span>
                                    ^^^^^^^^^^^^^^^ object type

src/Components/CaseCard/index.js:41
 41:             <span>Parliament session: {parliament_case.parliament_session.name}</span>
                                                            ^^^^^^^^^^^^^^^^^^ property `parliament_session`. Property not found in
 41:             <span>Parliament session: {parliament_case.parliament_session.name}</span>
                                            ^^^^^^^^^^^^^^^ object type

src/types.js:28
 28:   created: date,
                ^^^^ identifier `date`. Could not resolve name

src/utils/testFixtures.js:45
 45: export const Parties: Array<PartyType> = [
                                 ^^^^^^^^^ property `website`. Property not found in
       v
 46:   {
 47:     id: 2,
 48:     name: 'Vinsti Grænir',
 49:   },
       ^ object literal

src/utils/testFixtures.js:45
 45: export const Parties: Array<PartyType> = [
                                 ^^^^^^^^^ property `website`. Property not found in
       v
 50:   {
 51:     id: 1,
 52:     name: 'Framsókn',
 53:   },
       ^ object literal

```

## Todos
- [x] Fix type errors

### CODE REVIEW

**Comment by Ragnar-H** [comment]
I really recommend reading through https://flow.org/en/docs/types/ before reviewing this PR

**Comment by Ragnar-H** [comment]
@AriHrannar you can review this commit-by-commit

File: src/types.js [hunk 1]
```diff
@@ -19,12 +19,24 @@ export type PartyType = {
...
+export type ParliamentType = {
+  id: number,
+  name: string,
+};
+
+export type CaseType = {
+  id: number,
   case_status: string,
   case_type: string,
-  created: date,
```
Comment (Ragnar-H) [inline src/types.js]: `date` is from TypeScript.
Flow isn't as smart :(

File: src/types.js [hunk 2]
```diff
@@ -19,12 +19,24 @@ export type PartyType = {
...
-  modified: date,
+  created: string,
+  name: string,
+  number: number,
+  modified: string,
+  parliament_session: ParliamentSessionType,
```
Comment (Ragnar-H) [inline src/types.js]: Note how breaking down nested data structures from backend increases readibility

File: src/utils/testFixtures.js [hunk 3]
```diff
@@ -46,9 +46,11 @@ export const Parties: Array<PartyType> = [
   {
     id: 2,
     name: 'Vinsti Grænir',
+    website: 'https://vg.is',
```
Comment (Ragnar-H) [inline src/utils/testFixtures.js]: `PartyCard` component would fail without this parameter

File: .eslintrc [hunk 4]
```diff
@@ -115,7 +115,6 @@
     "no-undef-init": 2,
     "no-undef": 1,
     "no-unused-vars": 1,
-    "no-use-before-define": 1,
```
Comment (AriHrannar) [inline .eslintrc]: Why kill this?

File: src/types.js (hunk 1 above)
Comment (AriHrannar) [inline src/types.js]: aww

File: src/types.js (hunk 2 above)
Comment (AriHrannar) [inline src/types.js]: :+1:

File: .eslintrc (hunk 4 above)
Comment (Ragnar-H) [inline .eslintrc]: I had a look at http://eslint.org/docs/rules/no-use-before-define
Where it's not relevant to ES6 anymore. (also mentioned this in my commit message)

This rule gets really annoying when writing types.

**Pull request merged**